### ISSUE TITLE
AddScruber by file name/path

### ISSUE THREAD

**Issue by JanKrivanek** [VerifyTests/Verify#673]
When verifying multiple files at once with `VerifyDirectory` I sometimes have a need to scrub specific content just in a specific file, that's not determined only by it's extension.
I'm using scrubber by extension (via [`AddScrubber`](https://github.com/VerifyTests/Verify/blob/main/src/Verify/SettingsTask_Scrubbing.cs#L17)), but it would be nice to be able to tailor its applicability to a specific file.

Not sure if this can be seen as more broadly usefull.
One way of achieving this might be to have the `ApplyScrubbers` to first try to match the registered scrubber against the current file extension, if not a match, then by it's name and then possibly even by relative path - although it would change the meaning of the 'extension' within the [`AddScrubber`](https://github.com/VerifyTests/Verify/blob/main/src/Verify/SettingsTask_Scrubbing.cs#L17) signature, it still should be fully back compatible.

**Comment by SimonCropp** [comment]
can u give me an example of something u want to scrub in on file, that you do t want scrubbed from another file

**Comment by JanKrivanek** [comment]
tl;dr;: It's not about broken scenario but rather about optimizing perf and resources consumption.

When having multiple files in a directory and in need of scrubbing a content of just a specific one - then defining scrubber per extension is not a blocker, but leads to unnecessary processing (repeated on each dev testing or CI cycle), which can quickly have multiplying effect with higher amount of tests that are part of automation. I do not have this quantified though.
In our specific case - our engine might be producing multiple files, while only some of them are supposed to have some content replacement logic - when then testing such an output with `VerifyDirectory`, the scrubbing is needed only for the files expected to be processed by special logic.
Our options now:
 * Run Verify per files (though we love the VerifyDirectory and would prefer it)
 * Run VerifyDirectory, while sticking to scrubbing all files per extensions (and spare some cycles on scrubbing files that do not need scrubbing)

I'm trying to see if there is other option I'm missing. Or if the approach with adjusting the `AddScrubber` and `ApplyScrubbers` sounds reasonable (I can try to stitch some draft PR to discuss closer), or if you have different opinion.

**Comment by SimonCropp** [comment]
how does this look https://github.com/VerifyTests/Verify/blob/main/docs/verify-directory.md#filescrubber

**Comment by JanKrivanek** [comment]
Thank you @SimonCropp - it looks perfect!!
I tried `18.0.0-beta.31` and it nicely matches to our scenario.
